zipline/gens/brokers/rh/robinhood.py

```diff
@@ -0,0 +1,1147 @@
+
+        res = self.session.get(self.endpoints['investment_profile'])
+        res.raise_for_status()  #will throw without auth
+        data = res.json()
+
+        return data
+
+
+    def instruments(self, symbol):
+        ''' Generates an instrument object. Currently this is only used for 
+        placing orders, and generating and using the instrument object are handled
+        for you, so you can ignore this method'''
+        res = self.session.get(self.endpoints['instruments'], params={'query':symbol.upper()})
+        if res.status_code == 200:
+            return res.json()['results']
+        else:
+            raise Exception("Could not generate instrument object: %s " % res.headers)
+
+
+    def quote_data(self, stock=''):
+        """Fetch stock quote
+
+            Args:
+                stock (str): stock ticker, prompt if blank
+
+            Returns:
+                (:obj:`dict`): JSON contents from `quotes` endpoint
+        """
+
+        url = None
+
+        if stock.find(',') == -1:
+            url = str(self.endpoints['quotes']) + str(stock) + "/"
+        else:
+            url = str(self.endpoints['quotes']) + "?symbols=" + str(stock)
+
+        #Check for validity of symbol
+        try:
+            req = requests.get(url)
+            req.raise_for_status()
+            data = req.json()
+        except requests.exceptions.HTTPError:
+            raise RH_exception.InvalidTickerSymbol()
+
+
+        return data
+
+
+    # We will keep for compatibility until next major release
+    def quotes_data(self, stocks):
+        """Fetch quote for multiple stocks, in one single Robinhood API call
+
+            Args:
+                stocks (list<str>): stock tickers
+
+            Returns:
+                (:obj:`list` of :obj:`dict`): List of JSON contents from `quotes` endpoint, in the
+                    same order of input args. If any ticker is invalid, a None will occur at that position.
+        """
+
+        url = str(self.endpoints['quotes']) + "?symbols=" + ",".join(stocks)
+
+        try:
+            req = requests.get(url)
+            req.raise_for_status()
+            data = req.json()
+        except requests.exceptions.HTTPError:
+            raise RH_exception.InvalidTickerSymbol()
+
+
+        return data["results"]
+
+
+    def get_quote_list(self, 
+                       stock='', 
+                       key=''):
+        """Returns multiple stock info and keys from quote_data (prompt if blank)
+
+            Args:
+                stock (str): stock ticker (or tickers separated by a comma)
+                , prompt if blank
+                key (str): key attributes that the function should return
+
+            Returns:
+                (:obj:`list`): Returns values from each stock or empty list
+                               if none of the stocks were valid
+
+        """
+
+        #Creates a tuple containing the information we want to retrieve
+        def append_stock(stock):
```

Comment:
I don't understand the purpose of this function, documentation does not help either. Could you please rename it to have a more meaningful name?